### PR TITLE
Sorr issue428 tools profiling gdoc to md

### PR DESCRIPTION
**Pull request by Ro0k1e**
Issue [#428](https://github.com/sorrentum/sorrentum/issues/428)
Continued on [PR #435](https://github.com/sorrentum/sorrentum/pull/435)
Closed PR#435 because want to update the branch to latest master

### CODE REVIEW

File: docs/Profiling.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+<!-- tocstop -->
+
+# Profiling end-to-end a command line
+
+- You can use the time-tested Linux `time` command to profile both time and memory
+
+```bash
+> /usr/bin/time -v COMMAND 2>&1 | tee time.log
+
+Command being timed: "...COMMAND..."
```
Comment (DanilYachmenev) [inline docs/Profiling.md:30]: let's indent each code block so they are being initiated from the same ident as the bullet lists they are related to
```
- You can use ...
    ```bash
    > ...
    ...
```

everywhere in the doc

File: docs/Profiling.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+## line_profiler
+
+- cProfile allows to break down the execution time into function calls, while
+kernprof allows to profile a function line by line.
+
+- GitHub: [<span
+class="underline">https://github.com/pyutils/line_profiler</span>](https://github.com/pyutils/line_profiler)
+
+- Install with: `pip install line_profiler`
```
Comment (DanilYachmenev) [inline docs/Profiling.md:164]: do it like for `pytest-profiling` block for consistency

```
- Install it with
    ```bash
    > pip install line_profiler
    ```
```

**Review by DanilYachmenev — COMMENTED** [review]
almost there, just a couple of formatting nits and ready to merge

**Comment by DanilYachmenev** [comment]
@Ro0k1e change label to `PR_for_reviewers` when you want us to review your work
otherwise, it can be omitted

**Comment by Ro0k1e** [comment]
> Only add the label when all the conversation are resolved and you want us to review your PR. RN I can see the labels are changed but no changes in your PR.

Understood. Thank you! Just made some changes in my PR and waiting for review!

**Comment by Ro0k1e** [comment]
> @Ro0k1e change label to `PR_for_reviewers` when you want us to review your work otherwise, it can be omitted

Understood! Thank you!

**Comment by DanilYachmenev** [comment]
pls fix the conflict

**Comment by Ro0k1e** [comment]
> pls fix the conflict

Thank you! I guess the master has been updated. Conflict is fixed now.

**Comment by samarth9008** [comment]
Resolve all the conversations as you make changes and change the labels accordingly.

File: docs/Profiling.md (hunk 2 above)
Comment (Ro0k1e) [reply docs/Profiling.md]: Updated

File: docs/Profiling.md (hunk 1 above)
Comment (Ro0k1e) [reply docs/Profiling.md]: Updated everywhere in the doc.

**Review by DanilYachmenev — COMMENTED** [review]
lgtm
but pls change label accordingly so we can see that it is time to review to work

File: docs/Profiling.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+  - top-level statistics
+
+  - plotting the call-graph
+
+  - custom statics
+
+## line_profiler
+
+- cProfile allows to break down the execution time into function calls, while
```
Comment (gpsaggese) [inline docs/Profiling.md:151]: Was the linter run?

File: docs/Profiling.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+kernprof allows to profile a function line by line.
+
+- GitHub: [<span
+class="underline">https://github.com/pyutils/line_profiler</span>](https://github.com/pyutils/line_profiler)
+
+- Install with: 
+	```bash
```
Comment (gpsaggese) [inline docs/Profiling.md:158]: Also here...

**Review by gpsaggese — COMMENTED** [review]
Can we run the linter and merge?

**Comment by samarth9008** [comment]
@DanilYachmenev Can you pls merge it? I ran the linter.

**Pull request merged**